src/NUnitFramework/framework/Internal/Execution/EventListenerTextWriter.cs

```diff
@@ -68,10 +68,14 @@ private bool TrySendToListener(string text)
             if (context == null || context.Listener == null)
                 return false;
 
-            string testName = context.CurrentTest != null
-                ? context.CurrentTest.FullName
-                : null;
-            context.Listener.TestOutput(new TestOutput(text, _streamName, testName));
+            string testName = null;
+            string testId = null;
+            if (context.CurrentTest != null)
+            {
+                testName = context.CurrentTest.FullName;
+                testId = context.CurrentTest.Id;
+            }
+            context.Listener.TestOutput(new TestOutput(text, _streamName, testId, testName));
```

Comment:
Could simplify this using ?. and ?? but it's not essential for merging as it's a stylistic decision.

```C#
var test = context.CurrentTest ?? null;
context.Listener.TestOutput(new TestOutput(text, _streamName, test?.Id, test?.FullName);
```

Reply:
Good suggestion